test/monitor_test.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import absolute_import, division, print_function
+
+import multiprocessing
+import subprocess
+import time
+import unittest
+
+import ray
+
+
+class MonitorTest(unittest.TestCase):
```

Comment:
This is a really nice test :) I didn't think of doing it that way. My previous attempt to do this looked much less clean..

https://github.com/ray-project/ray/blob/6e9657e6968e01717c81e83631f5fc4a75dee8db/test/multi_node_test.py#L61-L87